app/handlers/sensor.handler.go

```diff
@@ -0,0 +1,175 @@
+package handlers
+
+import (
+	"database-ms/app/models"
+	sensorSrv "database-ms/app/services/sensor"
+	"database-ms/utils"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type SensorHandler struct {
+	snsr sensorSrv.SensorServiceInterface
+}
+
+func NewSensorAPI(sensorService sensorSrv.SensorServiceInterface) *SensorHandler {
+	return &SensorHandler{
+		snsr: sensorService,
+	}
+}
+
+func (h *SensorHandler) Create(c *gin.Context) {
+	var newSensor models.Sensor
+	c.BindJSON(&newSensor)
+	result := make(map[string]interface{})
+
+	// Check if sensor already exists
+	if h.snsr.IsSensorAlreadyExists(c.Request.Context(), newSensor.ThingID, newSensor.SID) {
+		result = utils.NewHTTPError(utils.SensorAlreadyExists)
+		utils.Response(c, http.StatusBadRequest, result)
+		return
+	}
+
+	// TODO Check if Thing exists
+
+	// Generate last updated
+	newSensor.LastUpdate = utils.CurrentTimeInMilli()
+
+	// Generate id
+	newSensor.ID = bson.NewObjectId()
+
+	err := h.snsr.Create(c.Request.Context(), &newSensor)
+	var status int
+	if err != nil {
```

Comment:
If would personally handle the success condition first. We are most interested in a success, error handling first is confusing. Just a semantics thing. When I read code, I care most about the core logic first, THEN how errors are handled.